src/base/bittorrent/session.h

```diff
@@ -129,6 +136,16 @@ namespace BitTorrent
         };
         Q_ENUM_NS(DiskIOType)
 
+        enum class DiskIOWriteMode : int
```

Comment:
Well, I don't stare at the enums to read code. In this case, having it sorted alphabetically is better for maintenance IMO.

Reply:
>So what do you mean by "maintenance", if not easy reading, understanding and modification of the code?

That there is a strong/fixed ordering instead of some loose concept of similarities for a bunch of enums declarations.

Reply:
> Your idea of grouping by similarities will get out of hand when there are a few of enums correlated to each other and then you can't really find a good order. It often leads to messy/incomprehensible order for the reader.

IMO, you propose to abandon convenience in the general case only because it may not be suitable in some extreme unlikely cases. Okay, let's not make a stumbling block out of this.

Reply:
>IMO, you propose to abandon convenience in the general case only because it may not be suitable in some extreme unlikely cases. 

Grouping by similarities also doesn't work when enums are orthogonal/independent to each other (which is quite common).
And in the situation of some enums are related and some are not, I can only imagine a disaster for the reader/new contributor.